### PR TITLE
[SPARK-25100][TEST][FOLLOWUP] Refactor test cases in `FileSuite` and `KryoSerializerSuite`

### PR DESCRIPTION
**Pull request by jiangxb1987**
### What changes were proposed in this pull request?

Refactor test cases added by https://github.com/apache/spark/pull/26714, to improve code compactness.

### How was this patch tested?

Tested locally.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/FileSuite.scala [hunk 1]
```diff
@@ -702,32 +702,39 @@ class FileSuite extends SparkFunSuite with LocalSparkContext {
...
+      .set("spark.serializer", classOf[KryoSerializer].getName)
+    sc = new SparkContext(conf)
+
+    // Prepare the input RDD
+    val pairRDD = sc.textFile(inputFilePath).map(x => (x, x))
 
+    // Test saveAsTextFile()
+    val outputFilePath1 = new File(tempDir, "/out1").getAbsolutePath
+    pairRDD.saveAsTextFile(outputFilePath1)
+    assert(sc.textFile(outputFilePath1).collect() === Array("(1,1)", "(2,2)", "(3,3)"))
```
Comment (jiangxb1987) [inline core/src/test/scala/org/apache/spark/FileSuite.scala]: We should ensure the content in the output file is correct.

File: core/src/test/scala/org/apache/spark/serializer/KryoSerializerSuite.scala [hunk 2]
```diff
@@ -363,16 +363,14 @@ class KryoSerializerSuite extends SparkFunSuite with SharedSparkContext {
...
+    // structure.
 
-    val partitionPaths: mutable.Set[String] = mutable.Set()
-    partitionPaths.add("test3")
+    val ser = new KryoSerializer(conf).newInstance()
+    val addedAbsPathFiles = Map("test1" -> "test1", "test2" -> "test2")
+    val partitionPaths = Set("test3")
 
-    val taskCommitMessage1 = new TaskCommitMessage(addedAbsPathFiles.toMap -> partitionPaths.toSet)
+    val taskCommitMessage1 = new TaskCommitMessage(addedAbsPathFiles -> partitionPaths)
```
Comment (jiangxb1987) [inline core/src/test/scala/org/apache/spark/serializer/KryoSerializerSuite.scala]: We only need to test the object returned by `HadoopMapReduceCommitProtocol.commitTask()`, thus no need to create a mutable map/set then convert it.

**Comment by jiangxb1987** [comment]
cc @deshanxiao @HeartSaVioR @dongjoon-hyun Please take a look when you have time, thanks!

**Comment by dongjoon-hyun** [comment]
Thank you for making a followup and pinging me, @jiangxb1987 !

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. (This is a logical refactoring.)

**Review by HeartSaVioR — APPROVED** [review]
LGTM, this patch adds missing verification of content of output file as well as simplifies the code a bit.

**Review by deshanxiao — APPROVED** [review]
LGTM, thanks @jiangxb1987 for the verification of output!

**Comment by dongjoon-hyun** [comment]
Thank you all. Merged to master.

**Comment by jiangxb1987** [comment]
Thanks @deshanxiao @HeartSaVioR @dongjoon-hyun !

**Pull request closed without merging**